### PR TITLE
Implement RFC-0002 BearerDid

### PR DESCRIPTION
**Pull request by KendallWeihe**
🚧

Implementation for #240 for rust and bindings, ready for kt binding

### CODE REVIEW

File: bindings/uniffi/Cargo.toml
Comment (nitro-neal) [inline bindings/uniffi/Cargo.toml:13]: I noticed you have an `_` above for the package name but a `-` for the file name

File: crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs
Comment (nitro-neal) [inline crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs:33]: so now we have to do Ed2219 ed25519_generator_generate and import private key?

**Review by nitro-neal — APPROVED** [review]
Awesome! I don't know why we have to use this uniffi-wrapper, I figure its for wrangling the data to be binded in a specific way that you can't do with just the one uniffi file

File: crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs
Comment (nitro-neal) [reply crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs:33]: or just the old school way now, pass a key manager into bearer did

Comment (KendallWeihe) [reply crates/web5/src/apid/crypto/key_managers/in_memory_key_manager.rs:33]: that's right, I added some details to the [Draft RFC PR](https://github.com/TBD54566975/web5-rs/pull/240/files)

> Removal of `InMemoryKeyManager`'s `METHOD generate_key_material()`: this is nonsensical since the key manager implementation stores keys in memory and does not expose the functionality to export keys. The only use case would be for short-lived ephermal use cases, in which case, the developer can use the `Ed25519Generator` to generate the keys and call `METHOD import_jwk()` or the new `CONSTRUCTOR(private_jwks: []Jwk)`.

We're moving fast to unblock development, but I'll still backfill reasoning in the RFCs

We'll probably still want to expose a generator function, but I realized I was getting ahead of myself with convenience things, so removing for now

**Comment by nitro-neal** [comment]
This brings up a good point that we will need to bring up sooner or later.

-I think- that since this changed the bindings and regenerated `web5.kt` that when someone pulls web5-kt it will be broken out of the box.

I think we need some kind of internal versioning.

I imagine `just bind` would create the `web5.kt` file in a different directory outside the `kt` folder. Maybe just outside in a `latest` folder.

We can make another `just` command which is `just update-kt` or something which will just copy the `latest/web5.kt` into the official kt bound folder, and at that point it would break and the dev would have to update the kt to fix the bindings.

**Comment by KendallWeihe** [comment]
> I don't know why we have to use this uniffi-wrapper, I figure its for wrangling the data to be binded in a specific way that you can't do with just the one uniffi file

That's right, and I think we can clean this up later, but for some reason I can't dig into right now, and I think it's a result of the custom UniFFI `build.rs` script, I can't consume the Rust code from the wrapper in the tbdex UniFFI crate without splitting it out into a dedicated crate. And yeah I've come to terms that we'll have to effectively wrap everything other than purely data structures, it's just how it be, but it's fine because it's a recipe to follow.

**Comment by KendallWeihe** [comment]
> This brings up a good point that we will need to bring up sooner or later.\
> 
> -I think- that since this changed the bindings and regenerated web5.kt that when someone pulls web5-kt it will be broken out of the box.
> 
> I think we need some kind of internal versioning.

Great point, I shouldn't actually generate the kt bindings, I should hold off for you to do it when you make the kt changes. Thanks, I'll keep this in mind next time!

File: bindings/uniffi/Cargo.toml
Comment (KendallWeihe) [reply bindings/uniffi/Cargo.toml:13]: Good catch, I don't like that inconsistency, I'll fix. Also I'm going to make the directory name match the exact rust crate name. So:

- `bindings/uniffi` --> `bindings/web5_uniffi`
- `bindings/uniffi-wrapper` --> `bindings/web5_uniffi_wrapper`

Comment (KendallWeihe) [reply bindings/uniffi/Cargo.toml]: done

**Comment by KendallWeihe** [comment]
> Great point, I shouldn't actually generate the kt bindings, I should hold off for you to do it when you make the kt changes. Thanks, I'll keep this in mind next time!

@nitro-neal actually I went ahead and restored `bound/` from `main` so this PR shouldn't break the existing kt project

**Pull request merged**
